include/deal.II/numerics/fe_field_function.templates.h

```diff
@@ -140,7 +141,14 @@ namespace Functions
     FEValues<dim> fe_v(mapping, cell->get_fe(), quad,
                        update_gradients);
     fe_v.reinit(cell);
-    fe_v.get_function_gradients(data_vector, gradients);
+
+    // Unfortunately we still need a temporary argument as we want to
+    // evaluate a gradient of a (generally) multicomponent function at
+    // a single quadrature point.
+    std::vector< std::vector<Tensor<1,dim,number> > > vgrads
+    (1,  std::vector<Tensor<1,dim,number> >(n_components) );
+    fe_v.get_function_gradients(data_vector, vgrads);
+    gradients = std::vector<Tensor<1,dim,number> >(vgrads[0].begin(), vgrads[0].end());
```

Comment:
Could we bypass the creation of the array in case we just have one component? (It is a coincidence that it works in that case because it formally mixes quadrature points with components in the `std::vector` arguments, but I like to avoid memory allocations if possible.)